.github/workflows/preview.yml

```diff
@@ -22,8 +22,9 @@ jobs:
         with:
           version: '1.10'
 
+      # Note: needs resolve() to fix #518
       - name: Instantiate Julia environment
-        run: julia --project=. -e 'using Pkg; Pkg.instantiate()'
+        run: julia --project=. -e 'using Pkg; Pkg.instantiate(); Pkg.resolve()'
```

Comment:
Hmm, `Pkg.resolve()` doesn't update packages though; that would be `Pkg.update()`.

I'm sure I'm missing some weird edge case of an edge case 😄 but the only situation I could think of where `resolve()` actually changes anything is the already edge case where CI runs with a Julia version different from that in the Manifest.

Reply:
Hmm, I found https://discourse.julialang.org/t/what-is-the-difference-between-pkg-resolve-and-pkg-instantiate/77643/2:

> resolve will first check that the Manifest.toml is consistent with the Project.tomls of the current project and all its dependencies, updating it if necessary, then instantiate.

> update will use the newest possible dependencies (subject to compatibility constraints)

But actually even better, I tested it locally :)

 - If Project.toml has no compat entry and Manifest has `Foo@0.4`, `resolve()` doesn't do anything, even if there's a newer version available

 - If Project.toml has a compat entry of `"Foo = 0.5"` and Manifest has `Foo@0.4`, resolve() will fail with an error. We shouldn't run into this case because in this repository there only has one compat entry for Turing itself, and we already check that the Turing version is consistent